tests/test_sql_query.py

```diff
@@ -1049,6 +1052,82 @@ async def test_sql_query_set_03(self):
         res = await self.squery_values('SHOW search_path;')
         self.assertEqual(res, [["public"]])
 
+    async def test_sql_query_set_04(self):
+        # database settings allow_user_specified_ids & apply_access_policies_sql
+        # should be unified over EdgeQL and SQL adapter
+
+        async def set_current_database(val: Optional[bool]):
+            if val is None:
+                await self.con.execute(
+                    f'''
+                    configure current database
+                        reset apply_access_policies_sql;
+                    '''
+                )
+            else:
+                await self.con.execute(
+                    f'''
+                    configure current database
+                        set apply_access_policies_sql := {str(val).lower()};
+                    '''
+                )
+
+            # I don't know why this is needed
+            import asyncio
+            await asyncio.sleep(0.5)
```

Comment:
No, it will show up in `cfg::DatabaseConfig` right away.

You need to put a retryloop on the query that actually depends on the setting being populated.

I'll have a fix up for the whole problem today, though

Reply:
https://github.com/edgedb/edgedb/pull/7865 should fix it